add Maze from video

YiKaiWu1125 opened the pull request:
解決 #4 新增新 影片版迷宮(Add new mazes from video)

YiKaiWu1125 committed:
parent e697cc860ee07e906d6e34712f1f104465a9b5bf
author YiKaiWu1125 <500000000kai@gmail.com> 1658766080 +0800
committer YiKaiWu1125 <500000000kai@gmail.com> 1658826095 +0800

Initial_Maze_from_video

add Maze from video

can_run_in_linux

initial_file

add maze form video

Revert "Merge branch 'MazeFromVideo' of https://github.com/YiKaiWu1125/ElecTrap_backend into MazeFromVideo"

This reverts commit 0cf2122, reversing
changes made to 6afd0da.

asas1asas200 commented:
Are you tested it on Linux?

```shell
$ python begin_video_game.py
Traceback (most recent call last):
  File "/home/zeng/Desktop/ElecTrap_backend/Mazes_from_video/begin_video_game.py", line 1, in <module>
    from asyncio.windows_events import NULL
  File "/usr/lib/python3.10/asyncio/windows_events.py", line 6, in <module>
    raise ImportError('win32 only')
ImportError: win32 only
```

asas1asas200 commented:
Consider use [autopep8](https://pypi.org/project/autopep8/) or other tools to format your python code.

asas1asas200 commented on Mazes_from_video/playvideo_and_getcoordinate.py:
```diff
@@ -0,0 +1,252 @@
+import time
+import numpy as np
+
+width = 1920
+hight = 1080
+half_track_width = 50
```
Be aware of the global variables usage.

asas1asas200 commented on Mazes_from_video/playvideo_and_getcoordinate.py:
```diff
@@ -0,0 +1,252 @@
+            if cv2.waitKey(5) & 0xFF == 27:
+                break
+    
+#right_hand,binarization_arr=get_track("test1.mp4")
+#print("successful end.")
+#print("right_hand")
+#print(right_hand)
+#print("binarization_arr")
+#for i in range(hight):
+#    for j in range(width):
+#        if (binarization_arr[1][j][i] == 1):
+#            print("x:" + str(j) + " y:" + str(i))
+#print("game over.")
+#print(len(binarization_arr))
+#print(len(right_hand))
+#print(len(binarization_arr[0][1]))
+
+#path = 'output.txt'
+#f = open(path, 'w')
+#for h in range (hight+5):
+#    for w in range (width+5):
+#        for i in range(len(binarization_arr)):
+#            k = 0
+#            if binarization_arr[i][w][h] == 1:
+#                k = 1
+#                break
+#        print(str(k),end='', file=f)
+#    print('', file=f)
+#f.close()
```
Remove this block if it'is not needed, otherwise add some descriptions about using it.

asas1asas200 commented on Mazes_from_video/playvideo_and_getcoordinate.py:
```diff
@@ -0,0 +1,209 @@
+def get_track(video_name):
+    binarization_arr = [] # [nwo_number][hight][width]
+    right_hand = []
+    cap = cv2.VideoCapture(str(video_name))
```
The `cv2.VideoCapture()` will be called twice causes warning message below:
```shell
[ WARN:0@7.633] global /io/opencv/modules/videoio/src/cap_v4l.cpp (902) open VIDEOIO(V4L2:/dev/video1): can't open camera by index
```

YiKaiWu1125 replied:
第一次為撥放放範例影片
第二次為開始由遊戲
的確會調用到兩次`cv2.VideoCapture()`
但在結束都有`cap.release()`，這樣可執行嗎?

YiKaiWu1125 committed:
已修復下列問題

已修復
 +  移除 from asyncio.windows_events import NULL
 +  移除 無須用的輸出(print)
 +  冗餘 commit 合併問題
 +  移除全域變數
 +  新增 在playvideo_and_getcoordinate 當鏡頭被手動終止時，也會關閉攝像頭
 +  can't open camera by index 已將初始cmaera_index 從`1`更改為`0`

YiKaiWu1125 commented:
已在新commit中修復
 +  移除 from asyncio.windows_events import NULL 
 +  移除 無須用的輸出(print)
 +  冗餘 commit 合併問題
 +  移除全域變數
 +  新增 在playvideo_and_getcoordinate 當鏡頭被手動終止時，也會關閉攝像頭

asas1asas200 commented:
Don't mention the problem as timestamp, descirbe that more specific.

The issue of `cv2.VideoCapture()` should be fixed.

YiKaiWu1125 commented:
已新增修復
 +  can't open camera by index 已將初始cmaera_index 從`1`更改為`0`

asas1asas200 committed:
restyle: format the code by black
